cxx-squid/src/main/java/org/sonar/cxx/CxxVCppBuildLogParser.java

```diff
@@ -324,155 +351,101 @@ private void ParseCommonCompilerOptions(String line, String fileElement) {
+
+    //_Wp64 Defined when specifying /Wp64. Deprecated in Visual Studio 2010 and Visual Studio 2012, and not supported starting in Visual Studio 2013
     if (line.contains("/Wp64 ")) {
       AddMacro("_Wp64", fileElement);    
     }   
-    
+
     //_M_AMD64 Defined for x64 processors.
     //_WIN32 Defined for applications for Win32 and Win64. Always defined.
     //_WIN64 Defined for applications for Win64.
     //_M_X64 Defined for x64 processors.    
     //_M_IX86 Defined for x86 processors. See the Values for _M_IX86 table below for more information. This is not defined for x64 processors.
     //_M_IA64 Defined for Itanium Processor Family 64-bit processors.
-    //_M_IX86_FP Expands to a value indicating which /arch compiler option was used:
-    //    0 if /arch was not used.    
-    //    1 if /arch:SSE was used.
-    //    2 if /arch:SSE2 was used.   
-    if(platform.equals("x64")) {
+    if(platform.equals("x64")||line.contains("/D WIN64")) {
+      // Defined for compilations that target x64 processors.
       AddMacro("_WIN32", fileElement);
+      // This is not defined for x86 processors.
       AddMacro("_WIN64", fileElement);
       AddMacro("_M_X64", fileElement);
       AddMacro("_M_IA64", fileElement);
       AddMacro("_M_AMD64", fileElement);
     }
-    
-    if(platform.equals("Win32")) {
+    else if(platform.equals("Win32")||line.contains("/D WIN32")) {
+      // Defined for compilations that target x86 processors. 
       AddMacro("_WIN32", fileElement);
-      AddMacro("_M_IX86", fileElement);
-      AddMacro("_M_IX86_FP", fileElement);
-    }        
-    
-    // Weird Stuff, context sensite //
-    //__cplusplus Defined for C++ programs only.
-    //__FUNCDNAME__ Valid only in a function. Defines the decorated name of the enclosing function as a string.
-    //__FUNCDNAME__ is not expanded if you use the /EP or /P compiler option.
-    //__FUNCSIG__ Valid only in a function. Defines the signature of the enclosing function as a string.
-    //__FUNCSIG__ is not expanded if you use the /EP or /P compiler option.       
-    //__FUNCTION__ Valid only in a function. Defines the undecorated name of the enclosing function as a string. __FUNCTION__ is not expanded if you use the /EP or /P compiler option.
-        
-    // DEPRECATED STUFF //
-    //_M_MPPC Defined for Power Macintosh platforms (no longer supported).
-    //_M_MRX000 Defined for MIPS platforms (no longer supported).
-    //_M_PPC Defined for PowerPC platforms (no longer supported).
-    //_M_ALPHA Defined for DEC ALPHA platforms (no longer supported).
-  }       
-
-  private void ParseV100CompilerOptions(String line, String fileElement) {
-    // _CPPUNWIND Defined for code compiled with /GX (Enable Exception Handling).
+      //This is not defined for x64 processors.
+      AddMacro("_M_IX86", fileElement); 
+    }
+    // VC++ 16.0
     if (line.contains("/GX ")) {
       AddMacro("_CPPUNWIND", fileElement);    

```

Comment:
The parameter for cl.exe is not present for newer VC++ compiler versions and therefore this is only relevant for VS2010. The definition will only be created once even if it is multiple defined. I moved it here to have the "_CPPUNWIND" setting at one location but his can be moved back without any Impact.